model/trigger/host.go

```diff
@@ -0,0 +1,169 @@
+
+import (
+	"bytes"
+	"fmt"
+	"text/template"
+	"time"
+
+	"github.com/evergreen-ci/evergreen"
+	"github.com/evergreen-ci/evergreen/model/event"
+	"github.com/evergreen-ci/evergreen/model/host"
+	"github.com/evergreen-ci/evergreen/model/notification"
+	"github.com/mongodb/grip/message"
+	"github.com/pkg/errors"
+)
+
+func init() {
+	registry.registerEventHandler(event.ResourceTypeHost, event.EventHostExpirationWarningSent, makeHostTriggers)
+}
+
+const (
+	objectHost        = "host"
+	triggerExpiration = "expiration"
+
+	// notification templates
+	expiringHostTitle = `{{.Distro}} host termination reminder`
+	expiringHostBody  = `Your {{.Distro}} host with id {{.ID}} will be terminated at {{.ExpirationTime}}. Visit {{.URL}} to extend its lifetime.`
+)
+
+type hostTemplateData struct {
+	ID             string
+	Distro         string
+	ExpirationTime time.Time
+	URL            string
+}
+
+func makeHostTriggers() eventHandler {
+	t := &hostTriggers{}
+	t.base.triggers = map[string]trigger{
+		triggerExpiration: t.hostExpiration,
+	}
+
+	return t
+}
+
+type hostTriggers struct {
+	event        *event.EventLogEntry
+	data         *event.HostEventData
+	host         *host.Host
+	templateData hostTemplateData
+
+	base
+}
+
+func (t *hostTriggers) Fetch(e *event.EventLogEntry) error {
+	var ok bool
+	var err error
+	t.data, ok = e.Data.(*event.HostEventData)
+	if !ok {
+		return errors.New("expected host event data")
+	}
+
+	t.host, err = host.FindOneId(e.ResourceId)
+	if err != nil {
+		return errors.Wrap(err, "failed to fetch host")
+	}
+	if t.host == nil {
+		return errors.New("couldn't find host")
+	}
+
+	settings, err := evergreen.GetConfig()
+	if err != nil {
+		return errors.Wrap(err, "error retrieving settings")
+	}
+
+	t.templateData = hostTemplateData{
+		ID:             t.host.Id,
+		Distro:         t.host.Distro.Id,
+		ExpirationTime: t.host.ExpirationTime,
+		URL:            fmt.Sprintf("%s/ui/spawn", settings.Ui.Url),
+	}
+	t.event = e
+	return nil
+}
+
+func (t *hostTriggers) Selectors() []event.Selector {
+	return []event.Selector{
+		{
+			Type: selectorID,
+			Data: t.host.Id,
+		},
+		{
+			Type: selectorObject,
+			Data: objectHost,
+		},
+		{
+			Type: selectorOwner,
+			Data: t.host.StartedBy,
+		},
+	}
+}
+
+func (t *hostTriggers) generate(sub *event.Subscription, subjectTempl, bodyTempl string) (*notification.Notification, error) {
+	var payload interface{}
+	var err error
+	switch sub.Subscriber.Type {
+	case event.EmailSubscriberType:
+		payload, err = hostEmail(t.templateData, subjectTempl, bodyTempl, sub.Selectors)
+	case event.SlackSubscriberType:
+		payload, err = hostSlackMessage(t.templateData, bodyTempl, sub.Selectors)
+	}
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to parse templates")
+	}
+
+	return notification.New(t.event, sub.Trigger, &sub.Subscriber, payload)
+}
+
+func hostEmail(t hostTemplateData, subjectString, bodyString string, selectors []event.Selector) (*message.Email, error) {
```

Comment:
Please either make this a member function, or prefix the function's name better (ex: hostExpirationEmailPayload)